test/tests.bats

```diff
@@ -1,3 +1,5 @@
+set -x
```

Comment:
Very dangerous for a file that is this large. Probably the reasons tests are unstable. This would need extensive testing. It also does not work in functions and will result in unexpected behavior.

Reply:
whoops -- thanks. Fixed!